### PR TITLE
Detect podman as separate container type

### PR DESCRIPTION
**Pull request by keszybz**

### CODE REVIEW

File: src/basic/virt.c [hunk 1]
```diff
@@ -471,6 +470,8 @@ int detect_container(void) {
                 /* If we are PID 1 we can just check our own environment variable, and that's authoritative. */
 
                 e = getenv("container");
+                if (!e)
+                        goto check_sched;
                 if (isempty(e)) {
```
Comment (poettering) [inline src/basic/virt.c:480]: the isempty() check kinda does the same thing, more or less... we should keep only one

File: src/basic/virt.c [hunk 2]
```diff
@@ -498,8 +499,9 @@ int detect_container(void) {
         if (r < 0) /* This only works if we have CAP_SYS_PTRACE, hence let's better ignore failures here */
                 log_debug_errno(r, "Failed to read $container of PID 1, ignoring: %m");
 
-        /* Interestingly /proc/1/sched actually shows the host's PID for what we see as PID 1. Hence, if the PID shown
-         * there is not 1, we know we are in a PID namespace. and hence a container. */
+        /* Interestingly /proc/1/sched actually shows the host's PID for what we see as PID 1. If the PID
+         * shown there is not 1, we know we are in a PID namespace and hence a container. */
+ check_sched:
```
Comment (poettering) [inline src/basic/virt.c:509]: this test doesn't work anyway anymore, does it? maybe it's time to remove this bit?

File: src/basic/virt.h [hunk 3]
```diff
@@ -31,6 +31,7 @@ enum {
         VIRTUALIZATION_LXC,
         VIRTUALIZATION_OPENVZ,
         VIRTUALIZATION_DOCKER,
+        VIRTUALIZATION_PODMAN,
```
Comment (poettering) [inline src/basic/virt.h:34]: please also add to the man pages, i.e. systemd-detect-virt and systemd.unit

File: src/basic/virt.c (hunk 1 above)
Comment (keszybz) [reply src/basic/virt.c:480]: It's on purpose, I wanted to treat the lack of the variable different than an empty assignment. I.e. undefined → do other checks, empty → no container, non-empty → container. WDYT?

File: src/basic/virt.c (hunk 2 above)
Comment (keszybz) [reply src/basic/virt.c:509]: It still works on older kernels, which we still support. I think we should keep it as long as any kernels where it works are reasonable to use.

**Comment by keszybz** [comment]
Updated:
- new comment about the isempty() check
- added missing docs

**Pull request merged**